addons/base_vat/models/res_partner.py

```diff
@@ -602,6 +606,10 @@ def check_vat_au(self, vat):
             return len(vat) == 11 and vat.isdigit()
         return check_func(vat)
 
+    def check_vat_t(self, vat):
+        if self.country_id.code == 'JP':
```

Comment:
Also i think that this line will be useless since it will go through that only if we are in japan anyway ?

Reply:
Not really, as explained above, the method that is called depends only on the country code at the start of the tax number as it the result of a simple concatenation of "check_vat_" + "[vat_number_country_code]" then passed into a getattr. This means that without this line, a Belgian company that would try to register a tax number starting with "T" would make the system call "check_vat_t" which would call "simple_vat_check('jp',vat)" which would return True if the code fits the japan standards. Regardless of the company's country.